zebrad/tests/acceptance.rs

```diff
@@ -1,30 +1,274 @@
+    let (mut cmd, guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
+
+    Ok((
+        cmd.stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn2()
+            .unwrap(),
+        guard,
+    ))
+}
+
+#[test]
+fn generate_no_args() -> Result<()> {
+    zebra_test::init();
+
+    let (child, _guard) = get_child_single_arg("generate")?;
+    let output = child.wait_with_output()?;
+    let output = output.assert_success()?;
+
+    output.stdout_contains(r"# Default configuration for zebrad.")?;
+
+    Ok(())
+}
+
+#[test]
+fn generate_args() -> Result<()> {
+    zebra_test::init();
+
+    // unexpected free argument `argument`
+    let (child, _guard) = get_child_multi_args(&["generate", "argument"])?;
+    let output = child.wait_with_output()?;
+    output.assert_failure()?;
+
+    // unrecognized option `-f`
+    let (child, _guard) = get_child_multi_args(&["generate", "-f"])?;
+    let output = child.wait_with_output()?;
+    output.assert_failure()?;
+
+    // missing argument to option `-o`
+    let (child, _guard) = get_child_multi_args(&["generate", "-o"])?;
+    let output = child.wait_with_output()?;
+    output.assert_failure()?;
 
-/// Executes your application binary via `cargo run`.
-pub static RUNNER: Lazy<CmdRunner> = Lazy::new(CmdRunner::default);
+    // Valid
+    let (child, _guard) = get_child_multi_args(&["generate", "-o", "file.yaml"])?;
+    let output = child.wait_with_output()?;
+    output.assert_success()?;
+
+    // Todo: Check if the file was created
+
+    Ok(())
+}
 
-/*
- * Disabled pending tracing config rework, so that merging abscissa fixes doesn't block on this
- * test failing because there's tracing output.
- *
-/// Example of a test which matches a regular expression
 #[test]
-fn version_no_args() {
-    let mut runner = RUNNER.clone();
-    let mut cmd = runner.arg("version").capture_stdout().run();
-    cmd.stdout().expect_regex(r"\A\w+ [\d\.\-]+\z");
+fn help_no_args() -> Result<()> {
+    zebra_test::init();
+
+    let (child, _guard) = get_child_single_arg("help")?;
+    let output = child.wait_with_output()?;
+    let output = output.assert_success()?;
+
+    output.stdout_contains(r"USAGE:")?;
+
+    Ok(())
+}
+
+#[test]
+fn help_args() -> Result<()> {
+    zebra_test::init();
+
+    // The subcommand "argument" wasn't recognized.
+    let (child, _guard) = get_child_multi_args(&["help", "argument"])?;
+    let output = child.wait_with_output()?;
+    output.assert_failure()?;
+
+    // option `-f` does not accept an argument
+    let (child, _guard) = get_child_multi_args(&["help", "-f"])?;
+    let output = child.wait_with_output()?;
+    output.assert_failure()?;
+
+    Ok(())
+}
+
+#[test]
+fn revhex_no_args() -> Result<()> {
+    zebra_test::init();
+
+    let (mut child, _guard) = get_child_single_arg("revhex")?;
+
+    //Program is waiting for input, we just exit after 1 second
+    std::thread::sleep(Duration::from_secs(1));
+    child.kill()?;
+
+    Ok(())
+}
```

Comment:
I don't think this test is worth it as is, I'd prefer to just remove this entirely rather than only testing that we can spawn and send a kill signal to a command without an error.

Reply:
Yea, agree. Thanks for making it.